e2e/test/iothub/messaging/FaultInjectionPoolAmqpTests.MessageReceiveFaultInjectionPoolAmqpTests.cs

```diff
@@ -854,151 +140,54 @@ public async Task Message_DeviceSak_GracefulShutdownReceiveUsingCallbackRecovery
-                testDeviceCallbackHandlers.ForEach(testDeviceCallbackHandler => testDeviceCallbackHandler.Dispose());
-            }
-
-            await FaultInjectionPoolingOverAmqp
-                .TestFaultInjectionPoolAmqpAsync(
-                    MessageReceive_DevicePrefix,
-                    transport,
-                    proxyAddress,
-                    poolSize,
-                    devicesCount,
-                    faultType,
-                    reason,
-                    delayInSec == TimeSpan.Zero ? FaultInjection.DefaultFaultDelay : delayInSec,
-                    durationInSec == TimeSpan.Zero ? FaultInjection.DefaultFaultDuration : durationInSec,
-                    (d, t, h) => { return Task.FromResult(false); },
-                    TestOperationAsync,
-                    CleanupOperationAsync,
-                    authScope,
-                    Logger)
-                .ConfigureAwait(false);
-        }
-
         private async Task ReceiveMessageUsingCallbackRecoveryPoolOverAmqpAsync(
-            TestDeviceType type,
             Client.TransportType transport,
             int poolSize,
             int devicesCount,
             string faultType,
-            string reason,
-            TimeSpan delayInSec = default,
-            TimeSpan durationInSec = default,
-            ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device,
-            string proxyAddress = null)
+            string reason)
         {
             // Initialize the service client
-            var serviceClient = ServiceClient.CreateFromConnectionString(TestConfiguration.IoTHub.ConnectionString);
+            using var serviceClient = ServiceClient.CreateFromConnectionString(TestConfiguration.IotHub.ConnectionString);
 
             async Task InitOperationAsync(DeviceClient deviceClient, TestDevice testDevice, TestDeviceCallbackHandler testDeviceCallbackHandler)
             {
+                await deviceClient.OpenAsync().ConfigureAwait(false);
                 await testDeviceCallbackHandler.SetMessageReceiveCallbackHandlerAsync().ConfigureAwait(false);
             }
 
             async Task TestOperationAsync(DeviceClient deviceClient, TestDevice testDevice, TestDeviceCallbackHandler testDeviceCallbackHandler)
             {
-                var timeout = TimeSpan.FromSeconds(20);
-                using var cts = new CancellationTokenSource(timeout);
-
                 (Message msg, string payload, string p1Value) = MessageReceiveE2ETests.ComposeC2dTestMessage(Logger);
                 testDeviceCallbackHandler.ExpectedMessageSentByService = msg;
                 await serviceClient.SendAsync(testDevice.Id, msg).ConfigureAwait(false);
-                Logger.Trace($"{nameof(FaultInjectionPoolAmqpTests)}: Sent message to device {testDevice.Id}: payload='{payload}' p1Value='{p1Value}'");
 
-                Client.Message receivedMessage = await deviceClient.ReceiveAsync(timeout).ConfigureAwait(false);
```

Comment:
I think this call was a mistake. I know I touched this code not too long ago, so maybe I messed it up, but either way I'm fixing it now.

All we need is to subscribe to the callback (line 166 in updated file), using the callback handler.